src/docs/arch-checkpoint.dox

```diff
@@ -0,0 +1,87 @@
+/*! @arch_page arch-checkpoint Checkpoint
+
+# Overview #
+
+A checkpoint is a known point in time from which WiredTiger can recover in the event of a
+crash or unexpected shutdown. WiredTiger checkpoints are created either via the API
+WT_SESSION::checkpoint, or internally. Internally checkpoints are created on startup, shutdown
+and during compaction.
+
+A checkpoint is performed within the context of snapshot isolation transaction as such the
+checkpoint has a consistent view of the database from beginning to end. The checkpoint can be run
+with or without a timestamp, generally a timestamp is used. This timestamp is not used as a
+\c read_timestamp for the checkpoint transaction. The checkpoint timestamp will be set to the
+current \c stable_timestamp during checkpoint prepare. This timestamp is written out with the
```

Comment:
Questions for these few sentences.  There are many timestamps.  When referring to a timestamp, give it a name, and refer to the same name later.

- The `checkpoint timestamp will be set`.  Is the the `read_timestamp` for the checkpoint transaction from the previous sentence?  If so, we should use the same `read_timestamp`.  Or maybe it's the timestamp from the API call?
- `This timestamp is not used as a \c read_timestamp`...  If not, what is it used for?

Reply:
I agree that is confusing, I'll try and be more clear.

Reply:
I've reworded it now, let me know what you think